pkg/seedadmission/pods.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seedadmission
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenlogger "github.com/gardener/gardener/pkg/logger"
+
+	"github.com/sirupsen/logrus"
+	"gomodules.xyz/jsonpatch/v2"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//MutatePod mutates pod if mutation conditions are met
+func MutatePod(ctx context.Context, c client.Client, logger *logrus.Logger, request *admissionv1beta1.AdmissionRequest) ([]jsonpatch.JsonPatchOperation, error) {
+	// Ignore all resources other than our expected ones
+	switch request.Resource {
+	case
+		metav1.GroupVersionResource{Group: corev1.SchemeGroupVersion.Group, Version: corev1.SchemeGroupVersion.Version, Resource: "pods"}:
+	default:
+		return nil, nil
+	}
+
+	obj, err := getRequestObject(ctx, c, request)
+	if err != nil {
+		return nil, client.IgnoreNotFound(err)
+	}
+
+	entryLogger := gardenlogger.
+		NewFieldLogger(logger, "resource", fmt.Sprintf("%s/%s/%s", request.Kind.Group, request.Kind.Version, request.Kind.Kind)).
+		WithField("operation", request.Operation).
+		WithField("namespace", request.Namespace)
+
+	entryLogger.Info("Handling request")
+	patch, err := admitMutatePodAndMutate(ctx, c, entryLogger, obj, request.Namespace)
+	if err != nil {
+		logger.Errorf("Error mutating pod %s/%s: %v", request.Name, request.Namespace, err)
+		return nil, err
+	}
+
+	return patch, nil
+}
+
+// admitMutatePod checks if the object mutation is confirmed and return the pathces needed. If the given object is a list of objects then it
```

Comment:
```suggestion
// admitMutatePod checks if the object mutation is allowed and return the patches needed. If the given object is a list of objects then it
```

Reply:
Can it even happen that the object is a list of objects?

Reply:
I don't know. I saw it from the extensions_crds.go and just copied the logic.

Reply:
Probably it cannot happen. I think that's special with `DELETE` operation?! cc @timuthy @mvladev @vpnachev @ialidzhikov

Reply:
I'm not aware of it either. Also `kubectl` for example transforms a `kubectl create -f list-of-shoots.yaml` to single `POST` requests.

Reply:
@rfranzke  Should I remove the List processing option?

Reply:
I'd remove it as it's code that can not be reached :) thanks!